### ISSUE TITLE
"Curated Genes" vs "Curated Genes (protein-coding)"  Gene models track labels Problem

### ISSUE THREAD

**Issue by Paul-Davis** [WormBase/website#3624]
It looks like something has changed/is misconfigured for the labels on the gene models in the "Curated Genes" vs "Curated Genes (protein-coding)"

http://staging.wormbase.org/tools/genome/gbrowse/c_elegans_PRJNA13758/?start=9051663;stop=9063267;ref=III;width=800;version=100;flip=0;grid=1;id=cd0c8f91c8e42eac125b7119702dfdf5;l=GENES_PSEUDOGENES%1EGENES_PROTEIN_CODING%1EGENES_NONCODING%1EGENES%1ELOCI%3Aoverview

"Curated Genes (protein-coding)" has the transcripts individually labelled with there sequence name and official gene name where available

"Curated Genes" only has the official gene name where available and the "Sequence_name" which is the same for all isoforms of that gene.

![coding genes](https://cloud.githubusercontent.com/assets/717707/6271704/77c10154-b85a-11e4-9e65-05894af5d418.png)

Could all of the Curated Genes tracks follow the "Curated Genes (protein-coding)" style? actually I think it's only the composite "Curated Genes"  track which is different, probably due to the additional complexity of aggregation.

There is an additional issue in the ncRNA track labels specifically for the miRNAs

Transcripts are 

a) Missing from both tracks

b) badly formatted and missing labels.

![mirna issue](https://cloud.githubusercontent.com/assets/717707/6271724/ac1b391a-b85a-11e4-9878-14ceb3ac3eac.png)

I think that the track config needs updating to pull all transcripts for a specific gene and also the aggregation looks a bit broken for the "Curated Genes"  track.

Thanks 

Paul

**Comment by scottcain** [comment]
I was planning on setting up my own GBrowse instance to work with this data, so I'll do that today and look into why data seems to be getting missed sometimes.  As an aside, I'd really like to get rid of the "protein coding" description attached to the mRNAs in the "Curated Genes (protein coding)" track, since it's fairly obvious that's what they are, given that's the track they're being displayed in.

**Comment by Paul-Davis** [comment]
yes, i guess that label takes up space that could be better used, it's probably only there as common elements were used to build multiple tracks.

let-7 example gff3 (as this is the most complicated miRNA with all 3 levels of annotation.)

```
 X       WormBase        gene    14744173        14744271        .       -       .       ID=Gene:WBGene00002285;Name=WBGene00002285;locus=let-7;sequence_name=C05G5.6;biotype=miRNA;Alias=let-7,C05G5.6
 X       miRNA_primary_transcript        miRNA_primary_transcript        14743590        14745321        .       -       .       ID=Transcript:C05G5.6.1;Parent=Gene:WBGene00002285;Name=C05G5.6.1
 X       miRNA_primary_transcript        miRNA_primary_transcript        14743590        14744480        .       -       .       ID=Transcript:C05G5.6.2;Parent=Gene:WBGene00002285;Name=C05G5.6.2
 X       miRNA_primary_transcript        exon    14743590        14744480        .       -       .       Parent=Transcript:C05G5.6.2
 X       miRNA_primary_transcript        miRNA_primary_transcript        14743590        14744296        .       -       .       ID=Transcript:C05G5.6.3;Parent=Gene:WBGene00002285;Name=C05G5.6.3
 X       miRNA_primary_transcript        exon    14743590        14744296        .       -       .       Parent=Transcript:C05G5.6.3
 X       miRNA_primary_transcript        exon    14743590        14745321        .       -       .       Parent=Transcript:C05G5.6.1
 X       pre_miRNA       pre_miRNA       14744173        14744271        .       -       .       ID=Transcript:C05G5.6;Parent=Gene:WBGene00002285;Name=C05G5.6;locus=let-7
 X       pre_miRNA       exon    14744173        14744271        .       -       .       Parent=Transcript:C05G5.6
 X       WormBase        miRNA   14744191        14744212        .       -       .       ID=Transcript:C05G5.6b;Parent=Gene:WBGene00002285;Name=C05G5.6b;locus=let-7
 X       WormBase        exon    14744191        14744212        .       -       .       Parent=Transcript:C05G5.6b
 X       WormBase        miRNA   14744234        14744255        .       -       .       ID=Transcript:C05G5.6a;Parent=Gene:WBGene00002285;Name=C05G5.6a;locus=let-7
 X       WormBase        exon    14744234        14744255        .       -       .       Parent=Transcript:C05G5.6a
```

**Comment by scottcain** [comment]
So part of what is going on with let-7 is that GBrowse doesn't know what to do with features of type 'miRNA_primary_transcript', but it "knows" they are there.  That's why in the curated genes track the label  starts so far to the left.  The glyph that is being used (the gene glyph, which in turn uses the processed_transcript glyph) knows to draw sub parts like exons, which works great when you're working with protein coding genes, where the beginning of the primary transcript is also the beginning of the final transcript (at least as far as it's annotated--I don't claim to know anything about the actual biology), which is to say, the beginning of the gene, the beginning of the processed transcript/mRNA and the beginning of the first exon are all at the same place, so the glyph makes a reasonable looking picture.  

The case with the miRNA transcript is different--here we have a large primary transcript that doesn't get drawn by itself and a few small exons somewhere in the middle that look like they are floating away from their gene label when they are drawn.  For comparison to the GBrowse image you have above, here is how the same region looks in JBrowse:

![screen shot 2015-02-20 at 12 02 57 pm](https://cloud.githubusercontent.com/assets/274751/6305698/1130f3ee-b8fb-11e4-892c-d53728ea2dbf.png)

This is better but still less than perfect; it ought to be connecting the two exon that are part of C05G5.6 but isn't. I don't know why yet, though I suspect it has to do with the glyph not knowing how to work with the subparts (it doesn't know what a pre_miRNA is), so it just draws everything.

**Comment by Paul-Davis** [comment]
Hi @scottcain 
The jbrowse view is almost spot on when it comes to the Curated Genes track, the placement of the let-7 label is a little confusing but this track essentially looks good.

The "Curated Genes (noncoding)" should mirrior what the "Curated Genes" track has for the let-7 transcripts as there doesn't need to be any linkage between the exons as these are 2 different mature products.

![mirna_jbrowse](https://cloud.githubusercontent.com/assets/717707/6325725/879a4636-bb42-11e4-9a9d-4c08fa94f82a.png)

**Comment by khowe** [comment]
This may be irrelevant to the display issue here, but the GFF3 posted above by @Paul-Davis appears wrong. All of those features should have source "WormBase". We will correct this in the WS247 GFF3.

**Comment by Paul-Davis** [comment]
@khowe has put a patched c.elegans file on the ftp, I have also deposited patched core species gff3 files.

**Comment by scottcain** [comment]
Hi @Paul-Davis I sort of forgot about this.  There current staging instance of GBrowse looks like this:

![screen shot 2015-10-12 at 11 59 05 am](https://cloud.githubusercontent.com/assets/274751/10432402/dc9b5f0c-70d8-11e5-8530-7b2c55ce0632.png)

And JBrowse looks like this:

![screen shot 2015-10-12 at 12 03 24 pm](https://cloud.githubusercontent.com/assets/274751/10432483/4a0ffaac-70d9-11e5-8a3b-9ac2f183377a.png)

Is that acceptable?  If so, please feel free to close.

**Comment by Paul-Davis** [comment]
Hi @scottcain , it's certainly much better, although the long transcripts are still missing from the noncoding track and the labelling is a bit hit and miss.

**Comment by scottcain** [comment]
OK, so I need to add miRNA_primary_transcript to the feature types that should be included in the non-coding genes tracks; that shouldn't be too hard :-)

**Comment by Paul-Davis** [comment]
;)

My understanding @scottcain is that the the "Curated Genes" track should just be a composite of: 
Curated Genes (noncoding)
Curated Genes (pseudogenes)
Curated Genes (protein-coding)
With the exception that the Curated Genes (protein-coding) also have the UTRs added via the Coding_transcript aggregator? I think this was done to prevent the expansion in models within the general  "Curated Genes" track and allow the users to see all possible predicted transcripts in the more detailed Curated Genes (protein-coding) track.

I think what you proposed in your previous comment will achieve this.
